src/test/java/com/google/step/youtube/YoutubeInfoScraperTest.java

```diff
@@ -14,132 +14,209 @@
-      Optional<String> actual = scraper.scrapeChannelUploadPlaylist(CHANNEL_ID);
-      assertThat(actual.isPresent(), equalTo(false));
-  }
-
-  @Test
-  public void scrapeChannelUploadPlaylist_channelExists() throws IOException {
-      mockChannelResponse = new ChannelListResponse();
-      Channel channel = new Channel();
-      channel.setContentDetails(new ChannelContentDetails().setRelatedPlaylists(
-          new RelatedPlaylists().setUploads(UPLOAD_ID)));
-      mockChannelResponse.setItems(Arrays.asList(channel));
-      when(mockListChannels.execute()).thenReturn(mockChannelResponse);
-      Optional<String> actual = scraper.scrapeChannelUploadPlaylist(CHANNEL_ID_THAT_EXISTS);
-      assertThat(actual.get(), equalTo(UPLOAD_ID));
-  }
-
-  @Test 
-  public void scrapeChannelUploadPlaylist_IOException() throws IOException {
-      when(mockListChannels.execute()).thenThrow(IOException.class);
-      assertThrows(IOException.class, () -> scraper.scrapeChannelUploadPlaylist(IOEXCEPTION));
-  }
-
-  @Test
-  public void scrapePlaylistItems_nonExistentUploadId() throws IOException {
-      mockPlaylistResponse = new PlaylistItemListResponse();
-      when(mockListPlaylistItems.execute()).thenReturn(mockPlaylistResponse.setItems(null));
-      Optional<List<PlaylistItem>> actual = scraper.scrapePlaylistItems(UPLOAD_ID_THAT_DOES_NOT_EXIST);
-      assertThat(actual.isPresent(), equalTo(false));
-  }
-
-  @Test
-  public void scrapePlaylistItems_emptyList() throws IOException {
-      mockPlaylistResponse = new PlaylistItemListResponse();
-      when(mockListPlaylistItems.execute()).thenReturn(mockPlaylistResponse.setItems(Arrays.asList()));
-      Optional<List<PlaylistItem>> actual = scraper.scrapePlaylistItems(UPLOAD_ID);
-      assertThat(actual.isPresent(), equalTo(false));
-  }
-
-  @Test
-  public void scrapePlaylistItems_uploadIdExists() throws IOException {
-      mockPlaylistResponse = new PlaylistItemListResponse();
-      mockPlaylistResponse.setItems(Arrays.asList(new PlaylistItem(), new PlaylistItem()));
-      when(mockListPlaylistItems.execute()).thenReturn(mockPlaylistResponse);
-      Optional<List<PlaylistItem>> actual = scraper.scrapePlaylistItems(UPLOAD_ID);
-      assertThat(actual.get(), equalTo(mockPlaylistResponse.getItems()));
-  }
-
-  @Test 
-  public void scrapePlaylistItems_IOException() throws IOException {
-      when(mockListPlaylistItems.execute()).thenThrow(IOException.class);
-      assertThrows(IOException.class, () -> scraper.scrapePlaylistItems(IOEXCEPTION));
-  }
+    private YoutubeInfoScraper scraper;
+    private ChannelListResponse mockChannelResponse;
+    private Channels.List mockListChannels;
+    private PlaylistItems.List mockListPlaylistItems;
+    private PlaylistItemListResponse mockPlaylistResponse;
+
+    private static final String CHANNEL_ID = "CHANNEL_ID";
+    private static final String CHANNEL_ID_THAT_DOES_NOT_EXIST = "CHANNEL_ID_THAT_DOES_NOT_EXIST";
```

Comment:
optional nit: I probably should've added this in the last review but EXISTENT.../NONEXISTENT... might be more concise.

Reply:
Done.